src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1644,19 +1753,19 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // return result obj which has error code and
                 // error description that is returned from
                 // server response
-                if (callbackHandle.callback != null) {
-                    callbackHandle.onSuccess(result);
-                }
                 return result;
             }
-        } else {
+        } 
+        else 
+        {
 
             // User is not using cache and explicitly
             // calling with refresh token. User should received
             // error code and error description in
             // Authentication result for Oauth errors
             Logger.v(TAG, "Cache is not used for Request:" + request.getLogInfo());
-            if (callbackHandle.callback != null) {
+            if (callbackHandle.callback != null) 
+            {
                 callbackHandle.onSuccess(result);
```

Comment:
I think callback handle checks whether callback is null, no need to double check it